core/deployment/src/main/java/io/quarkus/deployment/recording/BytecodeRecorderImpl.java

```diff
@@ -224,60 +227,66 @@ private static boolean isProxiable(Class<?> returnType) {
-                    storedMethodCalls.add(storedMethodCall);
-                    Class<?> returnType = method.getReturnType();
-                    if (method.getName().equals("toString")
-                            && method.getParameterTypes().length == 0
-                            && returnType.equals(String.class)) {
-                        return proxyName;
-                    }
-
-                    boolean voidMethod = method.getReturnType().equals(void.class);
-                    if (!voidMethod && !isProxiable(method.getReturnType())) {
-                        throw new RuntimeException("Cannot use " + method
-                                + " as a recorder method as the return type cannot be proxied. Use RuntimeValue to wrap the return value instead.");
-                    }
-                    if (voidMethod) {
-                        return null;
-                    }
-                    ProxyInstance instance = getProxyInstance(returnType);
-                    if (instance == null) {
-                        return null;
-                    }
+                }
+                StoredMethodCall storedMethodCall = new StoredMethodCall(theClass, method, args);
+                storedMethodCalls.add(storedMethodCall);
+                Class<?> returnType = method.getReturnType();
+                if (method.getName().equals("toString")
+                        && method.getParameterTypes().length == 0
+                        && returnType.equals(String.class)) {
+                    return proxy.getClass().getName();
+                }
 
-                    storedMethodCall.returnedProxy = instance.proxy;
-                    storedMethodCall.proxyId = instance.key;
-                    return instance.proxy;
+                boolean voidMethod = method.getReturnType().equals(void.class);
+                if (!voidMethod && !isProxiable(method.getReturnType())) {
+                    throw new RuntimeException("Cannot use " + method
+                            + " as a recorder method as the return type cannot be proxied. Use RuntimeValue to wrap the return value instead.");
+                }
+                if (voidMethod) {
+                    return null;
+                }
+                ProxyInstance instance = getProxyInstance(returnType);
+                if (instance == null) {
+                    return null;
                 }
 
-            });
+                storedMethodCall.returnedProxy = instance.proxy;
+                storedMethodCall.proxyId = instance.key;
+                return instance.proxy;
+            }
+
+        };
+
+        try {
+            if (recordingProxyFactories.containsKey(theClass)) {
+                return (T) recordingProxyFactories.get(theClass).newInstance(invocationHandler);
+            }
+            String proxyNameSuffix = "$$RecordingProxyProxy" + COUNT.incrementAndGet();
```

Comment:
It's minor but wouldn't we want a stable name here too?

Reply:
It does not really matter here, these don't go into the final application, and as all they do is record bytecode they don't show up in any profiles (as this is basically just adding to a list).